sdk/cosmos/azure-cosmos/test/test_user_configs.py

```diff
@@ -68,6 +68,12 @@ def test_enable_endpoint_discovery(self):
         self.assertTrue(client_default.client_connection.connection_policy.EnableEndpointDiscovery)
         self.assertTrue(client_true.client_connection.connection_policy.EnableEndpointDiscovery)
 
+    def test_authentication_error(self):
+        try:
+            cosmos_client.CosmosClient(url=_test_config.host, credential="wrong_key")
+        except exceptions.CosmosHttpResponseError as e:
+            self.assertEqual(e.status_code, http_constants.StatusCodes.UNAUTHORIZED)
```

Comment:
We should look at fixing this error experience. Customers should never have to check an HTTP status code, so if this is the only way to figure out that this was an authentication error, we should improve that :)